beacon-chain/sync/subscriber.go

```diff
@@ -125,14 +125,29 @@ func (r *RegularSync) subscribe(topic string, validate validator, handle subHand
 
 	// The main message loop for receiving incoming messages from this subscription.
 	messageLoop := func() {
+		go func() {
```

Comment:
again this is a message loop for each topic, so this routine will be launched multiple times. This would be best to be placed in when we register all our subscribers

Reply:
ok , will move it